groovy-language-server/src/test/java/com/palantir/ls/server/groovy/GroovyTreeParserTest.java

```diff
@@ -718,28 +582,328 @@ public void testFindReferences_includeDeclaration() throws IOException {
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 12, 12, false)));
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 14, 11, false)));
+
+        // FIELD references
+        expectedReferences = Sets.newHashSet(createLocation(file.toPath(), Ranges.createRange(11, 6, 11, 21)));
+        // Get references when providing definition position
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 1, 27, false)));
+        // Get references when providing position of usage
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 11, 15, false)));
+
+        // LOCAL variable references
+        expectedReferences = Sets.newHashSet(createLocation(file.toPath(), Ranges.createRange(13, 6, 13, 16)));
+        // Get references when providing definition position
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 10, 20, false)));
+        // Get references when providing position of usage
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 13, 10, false)));
+
+        // METHOD (non static) references
+        expectedReferences = Sets.newHashSet(
+                // TODO: figure out how to make these more precise
+                createLocation(file.toPath(), Ranges.createRange(11, 6, 11, 43)),
+                createLocation(file.toPath(), Ranges.createRange(12, 6, 12, 44)),
+                createLocation(file.toPath(), Ranges.createRange(13, 6, 13, 38)));
+        // Get references when providing definition position
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 3, 7, false)));
+        // Get references when providing position of usage
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 11, 23, false)));
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 12, 23, false)));
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 13, 23, false)));
+
+        // METHOD (static) references
+        expectedReferences = Sets.newHashSet(
+                // TODO: figure out how to make these more precise
+                createLocation(file.toPath(), Ranges.createRange(14, 6, 14, 35)));
+        // Get references when providing definition position
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 4, 15, false)));
+        // Get references when providing position of usage
+        assertEquals(expectedReferences, parser.findReferences(createReferenceParams(file.toURI(), 14, 20, false)));
+
+        // test2 references
+        expectedReferences = Sets.newHashSet(
+                // TODO: figure out how to make these more precise
```

Comment:
yes I did https://github.com/palantir/groovy-language-server/issues/124 I'll fix all the todos